### ISSUE TITLE
Prompt for inputs when not provided on the command line

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#38]
### TL;DR

Currently, the template variables provided by the user are given on the command line (e.g. `--inputs=database=postgres`). As a CLI user, I want to be prompted for any inputs that I didn't pass as flags. This is just for ease of use; as a CLI user I shouldn't have to know in advance what inputs are needed.

The library that we already use for CLI features, github.com/abcxyz/pkg/cli, has a Prompt() function that looks perfect for this: https://pkg.go.dev/github.com/abcxyz/pkg/cli#BaseCommand.Prompt

Example of how this could look:

```
$ abc templates render some-example-template

This template requires 3 inputs.

Input name: "num_worker_threads"
Required: false
Default: 10
Description: the number of concurrent workers that will serve HTTP requests.
Please enter a value for "num_worker_threads", or leave blank to use the default:
```

Idea for handling optional inputs: prompt for missing optional inputs (that were not set via `--input=foo=bar`), *unless* the user also specifies `--accept-default-inputs`. The `--accept-default-inputs` flag means that the user doesn't want to be bothered and they accept the defaults.

### Additional information

* Does our current Prompt function in pkg/cli support multi-arch multi-os?
* Developer experience around asking for inputs, messaging, how to default and how to incorporate into automation?

**Comment by verbanicm** [comment]
It would be good to think about what the prompt output looks like to provide the best developer experience.

**Comment by drevell** [comment]
This was built a long time ago, #137
